repo.go

```diff
@@ -872,16 +889,6 @@ func (r *Repo) AddOrUpdateSignature(roleFilename string, signature data.Signatur
 	signatures = append(signatures, signature)
 	s.Signatures = signatures
 
-	// Check signature on signed meta. Ignore threshold errors as this may not be fully
-	// signed.
-	for _, db := range dbs {
-		if err := db.VerifySignatures(s, role); err != nil {
-			if _, ok := err.(verify.ErrRoleThreshold); !ok {
```

Comment:
I'm not sure if you mean why did I delete this code, or why wasn't this code originally checking for `not verify.ErrRoleThreshold`.

For the first: I moved the signature check to above (`validSig`). Critically, instead of using this code that uses the key DB model, I use a single key verifier rather than the whole db. Otherwise, we can't detect whether the sig was valid, because now we don't error out on ONE invalid sig in the db verification function.

For the second: we were checking for errors that are _not_ `verify.ErrRoleThreshold`: see `!ok`

Reply:
Sorry, I meant the second meaning as you defined. Yes, agreed: doesn't make sense for checking non-threshold-related errors over the whole signatures collection when we are only adding/updating one sig here.